20_classes_lab/03.py

```diff
@@ -1,15 +1,12 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget constructor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works:
-
+#!/usr/bin/python
+class Widget(object):
+	def __init__(self, name):
+		self.name = name
+	def add_dependency(self, *args):
+		self.dependency = [arg.name for arg in args]
+	def build(self):
+		print self.dependency
+		
```

Comment:
I don't think it works, as some widgets are built multiple times. Try to fix the code so a widget is printed at most one time